### PR TITLE
Added docs on writing App tests

### PR DESCRIPTION
**Pull request by AverageMarcus**
### What this PR does / why we need it

Fixes https://github.com/giantswarm/giantswarm/issues/30647

Added more docs on how to test various different Apps with links to example test suites.

Viewing the new file directly might be the best way to view the contents: https://github.com/giantswarm/apptest-framework/blob/docs/docs/WRITING_TESTS.md

### Checklist

- [x] Updated CHANGELOG.md.

### Trigger E2E tests

Skipping tests as only docs changed.

### CODE REVIEW

File: docs/WRITING_TESTS.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+> Example: [tests/e2e/suites/defaultbundleapp](https://github.com/giantswarm/apptest-framework/blob/534f57426d183921e042e09cf6694ac2756d3862/tests/e2e/suites/defaultbundleapp/defaultbundleapp_suite_test.go)
+
+## Testing Default Apps
+
+> [!NOTE]
+> Testing of default apps is only supported with providers that have been updated to make use of Releases.
+
+Nothing special needs to be done to a test suite to make it compatible with a default app, instead the framework will detect at runtime if the app being tested is a default App or not by checking it against the Release spec for that provider.
+
+Once thing to be aware of when testing default apps is it's not possible to perform any actions before the installation of the App as it is done as part of the cluster creation.
```
Comment (mcharriere) [inline docs/WRITING_TESTS.md:125]:
```suggestion
One thing to be aware of when testing default apps is it's not possible to perform any actions before the installation of the App as it is done as part of the cluster creation.
```

**Review by mcharriere — APPROVED** [review]
small typo, otherwise looks good!

File: docs/WRITING_TESTS.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+## Testing Default Apps
+
+> [!NOTE]
+> Testing of default apps is only supported with providers that have been updated to make use of Releases.
+
+Nothing special needs to be done to a test suite to make it compatible with a default app, instead the framework will detect at runtime if the app being tested is a default App or not by checking it against the Release spec for that provider.
+
+One thing to be aware of when testing default apps is it's not possible to perform any actions before the installation of the App as it is done as part of the cluster creation.
+
+If testing an app within a bundle App that is a default App the framework will detect that bundle App from the Release and patch it to make sure the App being tested is installed as a child App during the cluster installtion phase.
```
Comment (stone-z) [inline docs/WRITING_TESTS.md:127]:
```suggestion
If testing an app within a bundle App that is a default App, the framework will detect that bundle App from the Release and patch it to make sure the App being tested is installed as a child App during the cluster installation phase.
```

File: docs/WRITING_TESTS.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+> [!TIP]
+> Example: [ingress-nginx-app - tests/e2e/suites/auth-bundle](https://github.com/giantswarm/ingress-nginx-app/tree/d3269ccf2e5d3cc044f9a4ea7c291c84806be75c/tests/e2e/suites/auth-bundle)
+
+## Adding New Test Cases
+
+Once [bootstrapped](https://github.com/giantswarm/apptest-framework#installation) your repo will have a test suite called `basic` that you can start adding tests to.
+
+There are 4 phases in which you can add tests:
+
+1. `AfterClusterReady` - These are run first, as soon as the workload cluster is deemed to be ready, and should be used to check for any needed pre-requisites in the cluster. This is optional and only need to be provided if you require some logic to run as soon as the cluster is stable.
```
Comment (stone-z) [inline docs/WRITING_TESTS.md:51]:
```suggestion
1. `AfterClusterReady` - These are run first, as soon as the workload cluster is deemed to be ready, and should be used to check for any needed pre-requisites in the cluster. This is optional and only need to be provided if you require some logic to run as soon as the cluster is stable. Note: Does not run for tests of default apps.
```

File: docs/WRITING_TESTS.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+1. `BeforeUpgrade` - These are only run if performing an upgrade tests and are run between installing the latest released version of your App and the version being tested. These are used to test that the App is in an expected state before performing the upgrade.
```
Comment (stone-z) [inline docs/WRITING_TESTS.md:52]:
```suggestion
1. `BeforeUpgrade` - These are only run if performing an upgrade tests and are run between installing the latest released version of your App and the version being tested. These are used to test that the App is in an expected state before performing the upgrade. Note: Does not run for tests of default apps.
```

**Review by stone-z — APPROVED** [review]
Looks very straightforward :+1: legitimately looking forward to trying it out

**Review by QuantumEnigmaa — APPROVED** [review]
I could not think of a better way to write this doc :)

**Pull request merged**